### PR TITLE
Add shufflenetv2 1.5 and 2.0 weights

### PR DESCRIPTION
**Pull request by YosuaMichael**
Resolve https://github.com/pytorch/vision/issues/3257

We train the model using similar with [the improved recipe](https://pytorch.org/blog/how-to-train-state-of-the-art-models-using-torchvision-latest-primitives/#the-training-recipe). Here are the commands:
```
# Choose the model
MODEL = shufflenet_v2_x1_5
# MODEL = shufflenet_v2_x2_0

# Training command
python \
    -u ~/script/run_with_submitit.py \
    --timeout 3000 --ngpus 8 --nodes 1 --batch-size=128 \
    --partition train --model $MODEL \
    --data-path="/datasets01_ontap/imagenet_full_size/061417" \
    --lr=0.5 --lr-scheduler=cosineannealinglr --lr-warmup-epochs=5 --lr-warmup-method=linear \
    --auto-augment=ta_wide --epochs=600 --random-erase=0.1 --weight-decay=0.00002 \
    --norm-weight-decay=0.0 --label-smoothing=0.1 --mixup-alpha=0.2 --cutmix-alpha=1.0 \
    --train-crop-size=176 --model-ema --val-resize-size=232 --ra-sampler --ra-reps=4
```

Once the training finished, we take the checkpoint of epoch with the highest `Acc@1` accuracy, for shufflenetv2_x2_0 we take epoch 595 and for shufflenetv2_x1_5 we take epoch 594, we take the non-ema models for both. Then, we test again the checkpoints with 1 gpu and batch_size=1, here are the commands and results:

```
# For shufflenetv2_x1_5
python -u ~/script/run_with_submitit.py \
    --timeout 3000 --nodes 1 --ngpus 1 --batch-size=1 \
    --partition train --model shufflenet_v2_x1_5 \
    --data-path="/datasets01_ontap/imagenet_full_size/061417" \
    --weights="ShuffleNet_V2_X1_5_Weights.IMAGENET1K_V1" \
    --test-only
# Test:  Acc@1 Acc@1 72.996 Acc@5 91.086

# For shufflenetv2_x2_0
python -u ~/script/run_with_submitit.py \
    --timeout 3000 --nodes 1 --ngpus 1 --batch-size=1 \
    --partition train --model shufflenet_v2_x1_5 \
    --data-path="/datasets01_ontap/imagenet_full_size/061417" \
    --weights="ShuffleNet_V2_X1_5_Weights.IMAGENET1K_V1" \
    --test-only
# Test:  Acc@1 76.230 Acc@5 93.006
```

We also provide quantized model using post training quantization. Here are the commands to do it:
```
# For shufflenet_v2_x1_5
python train_quantization.py --device='cpu' --post-training-quantize --backend='fbgemm' \
    --model=shufflenet_v2_x1_5 --weights="ShuffleNet_V2_X1_5_Weights.IMAGENET1K_V1" \
    --train-crop-size 176 --val-resize-size 232 --data-path /datasets01_ontap/imagenet_full_size/061417/

# For shufflenet_v2_x2_0
python train_quantization.py --device='cpu' --post-training-quantize --backend='fbgemm' \
    --model=shufflenet_v2_x1_5 --weights="ShuffleNet_V2_X1_5_Weights.IMAGENET1K_V1" \
    --train-crop-size 176 --val-resize-size 232 --data-path /datasets01_ontap/imagenet_full_size/061417/
```

And once we have the quantized model, we do evaluation with 1 gpu and batch_size=1. Here are the commands for evaluation and the corresponding result:

```
python train_quantization.py --device='cpu' --eval-batch-size=1 --test-only --backend='fbgemm' \
    --model='shufflenet_v2_x2_0' \
    --data-path="/datasets01_ontap/imagenet_full_size/061417" \
    --weights="ShuffleNet_V2_X2_0_QuantizedWeights.IMAGENET1K_FBGEMM_V1"
# Acc@1 72.052 Acc@5 90.700

python train_quantization.py --device='cpu' --eval-batch-size=1 --test-only --backend='fbgemm' \
    --model='shufflenet_v2_x1_5' \
    --data-path="/datasets01_ontap/imagenet_full_size/061417" \
    --weights="ShuffleNet_V2_X1_5_QuantizedWeights.IMAGENET1K_FBGEMM_V1"
# Acc@1 75.354 Acc@5 92.488
```

### CODE REVIEW

File: torchvision/models/shufflenetv2.py [hunk 1]
```diff
@@ -223,11 +223,37 @@ class ShuffleNet_V2_X1_0_Weights(WeightsEnum):
 
 
 class ShuffleNet_V2_X1_5_Weights(WeightsEnum):
-    pass
+    IMAGENET1K_V1 = Weights(
+        url="https://download.pytorch.org/models/shufflenetv2_x1_5-3c479a10.pth",
+        transforms=partial(ImageClassification, crop_size=224),
```
Comment (datumbox) [inline torchvision/models/shufflenetv2.py:228]: In the recipe you mentioned that you used `--val-resize-size=232` but that's not included here. Can you please add it and provide execution runs with the pre-trained weights that confirm the batch-size=1/gpu=1 accuracies?

Comment (YosuaMichael) [reply torchvision/models/shufflenetv2.py]: Thanks for noticing this! I forgot to add this and as a result the accuracy went down a bit for the eval. I have added this and update the resulting eval as well. The test execution run is on the PR description.

File: docs/source/models.rst [hunk 2]
```diff
@@ -499,8 +501,10 @@ Model                             Acc@1          Acc@5
 ================================  =============  =============
 MobileNet V2                      71.658         90.150
 MobileNet V3 Large                73.004         90.858
-ShuffleNet V2 x1.0                68.360         87.582
 ShuffleNet V2 x0.5                57.972         79.780
+ShuffleNet V2 x1.0                68.360         87.582
+ShuffleNet V2 x1.5                72.052         90.700
+ShuffleNet V2 x2.0                75.354         92.488
```
Comment (datumbox) [inline docs/source/models.rst:507]: These look good.

We should put the numbers for the quantized models as well below.

File: references/classification/README.md [hunk 3]
```diff
@@ -236,6 +236,18 @@ torchrun --nproc_per_node=8 train.py\
...
+torchrun --nproc_per_node=8 train.py \
+--batch-size=128 \
+--lr=0.5 --lr-scheduler=cosineannealinglr --lr-warmup-epochs=5 --lr-warmup-method=linear \
+--auto-augment=ta_wide --epochs=600 --random-erase=0.1 --weight-decay=0.00002 \
+--norm-weight-decay=0.0 --label-smoothing=0.1 --mixup-alpha=0.2 --cutmix-alpha=1.0 \
+--train-crop-size=176 --model-ema --val-resize-size=232 --ra-sampler --ra-reps=4
+```
+Here `$MODEL` is either `shufflenet_v2_x1_5` or `shufflenet_v2_x2_0`.
+
+
```
Comment (datumbox) [inline references/classification/README.md:250]: You need to explicitly specify how the x1.0 and x0.5 models were trained here. Not sure how much you can uncover from the recipe link. If not much we just need to say that it was contributed by a community member and place a link to the PR that added it for more info.

Also you will need to offer commands for the Quantization as well.

File: torchvision/models/quantization/shufflenetv2.py [hunk 4]
```diff
@@ -143,6 +152,42 @@ class ShuffleNet_V2_X1_0_QuantizedWeights(WeightsEnum):
...
+        url="https://download.pytorch.org/models/quantized/shufflenetv2_x1_5_fbgemm-d7401f05.pth",
+        transforms=partial(ImageClassification, crop_size=224, resize_size=232),
+        meta={
+            **_COMMON_META,
+            "recipe": "https://github.com/pytorch/vision/pull/5906",
+            "num_params": 3503624,
+            "unquantized": ShuffleNet_V2_X1_5_Weights.IMAGENET1K_V1,
+            "metrics": {
+                "acc@1": 72.052,
+                "acc@5": 90.700,
```
Comment (datumbox) [inline torchvision/models/quantization/shufflenetv2.py:166]: Looks good. Please offer the "training" commands for them and the equivalent validation commands for batch-size=1.

**Review by datumbox — COMMENTED** [review]
Thanks a lot @YosuaMichael. Just few minor comments but it's looking good.

File: docs/source/models.rst (hunk 2 above)
Comment (YosuaMichael) [reply docs/source/models.rst:507]: I think this is the section for quantized models already.

File: references/classification/README.md (hunk 3 above)
Comment (YosuaMichael) [reply references/classification/README.md:250]: Sure, will add the quantization command soon

**Review by datumbox — APPROVED** [review]
LGTM, thanks @YosuaMichael !
